File: src/13-degree-converter/typescript/src/main.ts
/*!
 * Polyglot Programming
 *
 * Copyright (c) 2024- Adam Burucs
 *
 * MIT Licensed
 */

type Celsius = number;
type Fahrenheit = number;

function intoFahrenheit(degree: Celsius): Fahrenheit {
  return ((degree * 9) / 5 + 32) as Fahrenheit;
}

function intoCelsius(degree: Fahrenheit): Celsius {
  return (((degree - 32) * 5) / 9) as Celsius;
}

function main() {
  console.log("Celsius and Fahrenheit degrees converter");
  console.log();

  let c25 = intoFahrenheit(25);
  console.log(`25 °C equals to ${c25} °F`);

  let f77 = intoCelsius(77);
  console.log(`77 °F equals to ${f77} °C`);
}

main();

File: src/13-degree-converter/rust/src/main.rs
/*!
 * Polyglot Programming
 *
 * Copyright (c) 2024- Adam Burucs
 *
 * MIT Licensed
 */

type Celsius = i16;
type Fahrenheit = i16;

fn into_fahrenheit(degree: Celsius) -> Fahrenheit {
    ((degree as f32 * 9.0 / 5.0) + 32.0) as Fahrenheit
}

fn into_celsius(degree: Fahrenheit) -> Celsius {
    ((degree as f32 - 32.0) * 5.0 / 9.0) as Celsius
}

fn main() {
    println!("Celsius and Fahrenheit degrees converter");
    println!();

    let c25 = into_fahrenheit(25);
    println!("25 °C equals to {c25} °F");

    let f77 = into_celsius(77);
    println!("77 °F equals to {f77} °C");
}
